main/main.c

```diff
@@ -271,49 +272,38 @@ static void php_disable_classes(void)
  */
 static void php_binary_init(void)
 {
-	char *binary_location = NULL;
+	zend_stat_t s;
+	char binary_location[MAXPATHLEN], *path;
 #ifdef PHP_WIN32
-	binary_location = (char *)malloc(MAXPATHLEN);
-	if (binary_location && GetModuleFileName(0, binary_location, MAXPATHLEN) == 0) {
-		free(binary_location);
-		PG(php_binary) = NULL;
+	if (GetModuleFileName(0, binary_location, MAXPATHLEN)) {
```

Comment:
Misses ==0 ?

Reply:
No. It should be `goto success;` instead, otherwise the success of that call will be treated as failure. Nice spot, though. Thanks :)